Filter design

ethan92429 opened the pull request:
Please update with master so that we can pull these changes in. I'll go through and do a review.

ethan92429 commented on aide_design/designs/civil_work/floc_classes/floc_concrete_chan.py:
```diff
@@ -0,0 +1,79 @@
+from aide_render.builder_classes import DP, HP
+
+
+class ConcreteChannels:
```
[Here's a post I made to ask this.](https://stackoverflow.com/questions/50533825/capture-function-declaration-in-a-python-decorator?noredirect=1#comment88079613_50533825)

ethan92429 commented on aide_design/designs/civil_work/floc.py:
```diff
@@ -0,0 +1,236 @@
+    >>> floc_design_dict = extract_types(my_floc, [DP], [dict])
+    >>> from aide_render.yaml import load, dump
+    >>> dump(floc_design_dict)
+    "BottomBaffles_Assembly:\n  BottomBaffle: {Height: !DP '1.728 meter', Thickness: !DP '2 millimeter', Width: !DP '0.3134\n      meter'}\n  EntTank_Length: !DP '2.2 meter'\n  Length: !DP '7.35 meter'\n  Num_Exit: !DP '18 '\n  Num_Inlet: !DP '26 '\n  Spacing: !DP '0.272 meter'\n  Thickness: !DP '2 millimeter'\n  TotalNum: !DP '2 '\n  WallThickness: !DP '0.15 meter'\n  Width: !DP '0.3134 meter'\n ...
```
How will we format this? The evaluation line is quite long and unruly to handle. Perhaps we can honor this using the new line characters?

fletchapin replied:
We discussed this on Friday. I'm not sure what we decided the solution was though.

fletchapin commented on aide_design/designs/civil_work/floc.py:
```diff
@@ -0,0 +1,236 @@
+
+
+class Flocculator:
+    """This is a flocculator class. It already has a lot of features built in,
```
We just updated the docstring to add more info on the design algorithm

fletchapin commented on .pytest_cache/v/cache/lastfailed:
```diff
@@ -0,0 +1,9 @@
+{
```
This is in the .gitignore and the file has been deleted

fletchapin commented on aide_design/designs/civil_work/ent_tank.py:
```diff
@@ -0,0 +1,40 @@
+    # overrides
+    def __init__(self, q_plant, depth_end, W_chan, bod=None):
+
+        if bod:
```
I agree these  are boiler plate code but I don't have time to work on it so we should merge as is and have this be "future work"

ethan92429 replied:
Please reference the "future work" issue with this in it here.

fletchapin commented on aide_design/designs/civil_work/ent_tank.py:
```diff
@@ -0,0 +1,40 @@
+from aide_design.shared.units import unit_registry as u
```
We have an open issue for updating docstrings of the classes, and have a good model docstring in the floc class. We wrote this class together with Sage and forgot to add a docstring, sorry!

fletchapin commented on aide_design/designs/civil_work/floc_classes/ent_floc_baffle_support.py:
```diff
@@ -0,0 +1,55 @@
+        self.numberbaffles = DP(num_baffles_chan_n.magnitude)
+        self.numberentbaffles = DP(num_baffles_chan_1.magnitude)
+
+        self.BottomBaffle = dict(vars(BottomBaffle(L_bottom_baffle,
```
The way that extract_types was working (at least for me) required it to be a ```dict``` or it won't recurse and only the parameters at the top level of the class are extracted

fletchapin commented on aide_design/designs/civil_work/floc_classes/floc_chan.py:
```diff
@@ -0,0 +1,58 @@
+
+    """
+
+    def __init__(self, num_chan, L_ent_tank_max, h_chan, L_sed, W_chan,
```
I intentionally made them inputs so that everything would be dependent on a call from the floc class in order to avoid having the same variables (e.g. L_ent_tank_max) defined in two different places

ethan92429 commented:
Cool! I just ran the tests and 10 tests are failing. We can pull this in, but the tests need to pass - `pytest git_repos/aide_design'. Here's the output I get:

============================= test session starts ==============================
platform darwin -- Python 3.6.4, pytest-3.4.2, py-1.5.2, pluggy-0.6.0
rootdir: /Users/ethankeller/git_repos/AguaClara/AIDE/aide_design, inifile:
plugins: cov-2.5.1
collected 147 items                                                            

tests/test_k_value_of_reductions_utility.py FFFFFF.....                  [  7%]
tests/test_lfom.py ....F..F..                                            [ 14%]
tests/test_physchem.py ...................................F............. [ 47%]
..................................................F..................... [ 96%]
...                                                                      [ 98%]
tests/test_pipedatabase.py .                                             [ 99%]
tests/test_pipeline_utility.py .                                         [100%]

=================================== FAILURES ===================================
________ KValuesCalculationTest.test_k_value_expansion_into_large_tank _________

self = <tests.test_k_value_of_reductions_utility.KValuesCalculationTest testMethod=test_k_value_expansion_into_large_tank>

    def test_k_value_expansion_into_large_tank(self):
>       self.assertAlmostEqual(k.k_value_expansion(pipe.OD(4), pipe.OD(400), 4 * u.L / u.s), 1.0110511331493719)

tests/test_k_value_of_reductions_utility.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:172: in wrapper
    result = func(*new_values, **kw)
aide_design/shared/utility.py:136: in wrapper
    result = func(*args, **kwargs)
aide_design/shared/k_value_of_reductions_utility.py:54: in k_value_expansion
    f = pc.fric(flow, id_entrance, NU, ROUGHNESS)
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:170: in wrapper
    new_values, values_by_name = converter(ureg, values, strict)
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:116: in _converter
    args_as_uc[ndx][0])
../../../../.local/lib/python3.6/site-packages/pint/registry.py:1214: in _convert
    return super(ContextRegistry, self)._convert(value, src, dst, inplace)
../../../../.local/lib/python3.6/site-packages/pint/registry.py:937: in _convert
    return super(NonMultiplicativeRegistry, self)._convert(value, src, dst, inplace)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pint.registry.UnitRegistry object at 0x10b3e65f8>, value = 1e-06
src = <UnitsContainer({'meter': 2.0, 'second': -1.0})>
dst = <UnitsContainer({'kilogram': 1.0, 'meter': -1.0, 'second': -1.0})>
inplace = False, check_dimensionality = True

    def _convert(self, value, src, dst, inplace=False, check_dimensionality=True):
        """Convert value from some source to destination units.
    
            :param value: value
            :param src: source units.
            :type src: UnitsContainer
            :param dst: destination units.
            :type dst: UnitsContainer
    
            :return: converted value
            """
    
        if check_dimensionality:
    
            src_dim = self._get_dimensionality(src)
            dst_dim = self._get_dimensionality(dst)
    
            # If the source and destination dimensionality are different,
            # then the conversion cannot be performed.
            if src_dim != dst_dim:
>               raise DimensionalityError(src, dst, src_dim, dst_dim)
E               pint.errors.DimensionalityError: Cannot convert from 'meter ** 2 / second' ([length] ** 2 / [time]) to 'kilogram / meter / second' ([mass] / [length] / [time])

../../../../.local/lib/python3.6/site-packages/pint/registry.py:708: DimensionalityError
__ KValuesCalculationTest.test_k_value_expansion_into_very_large_pipe_laminar __

self = <tests.test_k_value_of_reductions_utility.KValuesCalculationTest testMethod=test_k_value_expansion_into_very_large_pipe_laminar>

    def test_k_value_expansion_into_very_large_pipe_laminar(self):
>       self.assertAlmostEqual(k.k_value_expansion(pipe.OD(1), pipe.OD(400), 0.1 * u.L / u.s), 1.0216612503304363)

tests/test_k_value_of_reductions_utility.py:26: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:172: in wrapper
    result = func(*new_values, **kw)
aide_design/shared/utility.py:136: in wrapper
    result = func(*args, **kwargs)
aide_design/shared/k_value_of_reductions_utility.py:54: in k_value_expansion
    f = pc.fric(flow, id_entrance, NU, ROUGHNESS)
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:170: in wrapper
    new_values, values_by_name = converter(ureg, values, strict)
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:116: in _converter
    args_as_uc[ndx][0])
../../../../.local/lib/python3.6/site-packages/pint/registry.py:1214: in _convert
    return super(ContextRegistry, self)._convert(value, src, dst, inplace)
../../../../.local/lib/python3.6/site-packages/pint/registry.py:937: in _convert
    return super(NonMultiplicativeRegistry, self)._convert(value, src, dst, inplace)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pint.registry.UnitRegistry object at 0x10b3e65f8>, value = 1e-06
src = <UnitsContainer({'meter': 2.0, 'second': -1.0})>
dst = <UnitsContainer({'kilogram': 1.0, 'meter': -1.0, 'second': -1.0})>
inplace = False, check_dimensionality = True

    def _convert(self, value, src, dst, inplace=False, check_dimensionality=True):
        """Convert value from some source to destination units.
    
            :param value: value
            :param src: source units.
            :type src: UnitsContainer
            :param dst: destination units.
            :type dst: UnitsContainer
    
            :return: converted value
            """
    
        if check_dimensionality:
    
            src_dim = self._get_dimensionality(src)
            dst_dim = self._get_dimensionality(dst)
    
            # If the source and destination dimensionality are different,
            # then the conversion cannot be performed.
            if src_dim != dst_dim:
>               raise DimensionalityError(src, dst, src_dim, dst_dim)
E               pint.errors.DimensionalityError: Cannot convert from 'meter ** 2 / second' ([length] ** 2 / [time]) to 'kilogram / meter / second' ([mass] / [length] / [time])

../../../../.local/lib/python3.6/site-packages/pint/registry.py:708: DimensionalityError
_ KValuesCalculationTest.test_k_value_reduction_from_very_large_pipe_turbulent _

self = <tests.test_k_value_of_reductions_utility.KValuesCalculationTest testMethod=test_k_value_reduction_from_very_large_pipe_turbulent>

    def test_k_value_reduction_from_very_large_pipe_turbulent(self):
>       self.assertAlmostEqual(k.k_value_reduction(pipe.OD(400), pipe.OD(4), 4 * u.L / u.s), 105560.31724275621)

tests/test_k_value_of_reductions_utility.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:172: in wrapper
    result = func(*new_values, **kw)
aide_design/shared/utility.py:136: in wrapper
    result = func(*args, **kwargs)
aide_design/shared/k_value_of_reductions_utility.py:113: in k_value_reduction
    f = pc.fric(flow, id_entrance, NU, ROUGHNESS)
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:170: in wrapper
    new_values, values_by_name = converter(ureg, values, strict)
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:116: in _converter
    args_as_uc[ndx][0])
../../../../.local/lib/python3.6/site-packages/pint/registry.py:1214: in _convert
    return super(ContextRegistry, self)._convert(value, src, dst, inplace)
../../../../.local/lib/python3.6/site-packages/pint/registry.py:937: in _convert
    return super(NonMultiplicativeRegistry, self)._convert(value, src, dst, inplace)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pint.registry.UnitRegistry object at 0x10b3e65f8>, value = 1e-06
src = <UnitsContainer({'meter': 2.0, 'second': -1.0})>
dst = <UnitsContainer({'kilogram': 1.0, 'meter': -1.0, 'second': -1.0})>
inplace = False, check_dimensionality = True

    def _convert(self, value, src, dst, inplace=False, check_dimensionality=True):
        """Convert value from some source to destination units.
    
            :param value: value
            :param src: source units.
            :type src: UnitsContainer
            :param dst: destination units.
            :type dst: UnitsContainer
    
            :return: converted value
            """
    
        if check_dimensionality:
    
            src_dim = self._get_dimensionality(src)
            dst_dim = self._get_dimensionality(dst)
    
            # If the source and destination dimensionality are different,
            # then the conversion cannot be performed.
            if src_dim != dst_dim:
>               raise DimensionalityError(src, dst, src_dim, dst_dim)
E               pint.errors.DimensionalityError: Cannot convert from 'meter ** 2 / second' ([length] ** 2 / [time]) to 'kilogram / meter / second' ([mass] / [length] / [time])

../../../../.local/lib/python3.6/site-packages/pint/registry.py:708: DimensionalityError
____________ KValuesCalculationTest.test_k_value_reduction_laminar _____________

self = <tests.test_k_value_of_reductions_utility.KValuesCalculationTest testMethod=test_k_value_reduction_laminar>

    def test_k_value_reduction_laminar(self):
>       self.assertAlmostEqual(k.k_value_reduction(pipe.OD(1), pipe.OD(0.5), 0.1 * u.L / u.s), 2.1802730749680945)

tests/test_k_value_of_reductions_utility.py:16: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:172: in wrapper
    result = func(*new_values, **kw)
aide_design/shared/utility.py:136: in wrapper
    result = func(*args, **kwargs)
aide_design/shared/k_value_of_reductions_utility.py:113: in k_value_reduction
    f = pc.fric(flow, id_entrance, NU, ROUGHNESS)
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:170: in wrapper
    new_values, values_by_name = converter(ureg, values, strict)
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:116: in _converter
    args_as_uc[ndx][0])
../../../../.local/lib/python3.6/site-packages/pint/registry.py:1214: in _convert
    return super(ContextRegistry, self)._convert(value, src, dst, inplace)
../../../../.local/lib/python3.6/site-packages/pint/registry.py:937: in _convert
    return super(NonMultiplicativeRegistry, self)._convert(value, src, dst, inplace)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pint.registry.UnitRegistry object at 0x10b3e65f8>, value = 1e-06
src = <UnitsContainer({'meter': 2.0, 'second': -1.0})>
dst = <UnitsContainer({'kilogram': 1.0, 'meter': -1.0, 'second': -1.0})>
inplace = False, check_dimensionality = True

    def _convert(self, value, src, dst, inplace=False, check_dimensionality=True):
        """Convert value from some source to destination units.
    
            :param value: value
            :param src: source units.
            :type src: UnitsContainer
            :param dst: destination units.
            :type dst: UnitsContainer
    
            :return: converted value
            """
    
        if check_dimensionality:
    
            src_dim = self._get_dimensionality(src)
            dst_dim = self._get_dimensionality(dst)
    
            # If the source and destination dimensionality are different,
            # then the conversion cannot be performed.
            if src_dim != dst_dim:
>               raise DimensionalityError(src, dst, src_dim, dst_dim)
E               pint.errors.DimensionalityError: Cannot convert from 'meter ** 2 / second' ([length] ** 2 / [time]) to 'kilogram / meter / second' ([mass] / [length] / [time])

../../../../.local/lib/python3.6/site-packages/pint/registry.py:708: DimensionalityError
________ KValuesCalculationTest.test_k_value_reduction_square_turbulent ________

self = <tests.test_k_value_of_reductions_utility.KValuesCalculationTest testMethod=test_k_value_reduction_square_turbulent>

    def test_k_value_reduction_square_turbulent(self):
>       self.assertAlmostEqual(k.k_value_reduction(pipe.OD(4), pipe.OD(2), 4 * u.L/u.s), 5.6677039356929662)

tests/test_k_value_of_reductions_utility.py:13: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:172: in wrapper
    result = func(*new_values, **kw)
aide_design/shared/utility.py:136: in wrapper
    result = func(*args, **kwargs)
aide_design/shared/k_value_of_reductions_utility.py:113: in k_value_reduction
    f = pc.fric(flow, id_entrance, NU, ROUGHNESS)
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:170: in wrapper
    new_values, values_by_name = converter(ureg, values, strict)
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:116: in _converter
    args_as_uc[ndx][0])
../../../../.local/lib/python3.6/site-packages/pint/registry.py:1214: in _convert
    return super(ContextRegistry, self)._convert(value, src, dst, inplace)
../../../../.local/lib/python3.6/site-packages/pint/registry.py:937: in _convert
    return super(NonMultiplicativeRegistry, self)._convert(value, src, dst, inplace)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pint.registry.UnitRegistry object at 0x10b3e65f8>, value = 1e-06
src = <UnitsContainer({'meter': 2.0, 'second': -1.0})>
dst = <UnitsContainer({'kilogram': 1.0, 'meter': -1.0, 'second': -1.0})>
inplace = False, check_dimensionality = True

    def _convert(self, value, src, dst, inplace=False, check_dimensionality=True):
        """Convert value from some source to destination units.
    
            :param value: value
            :param src: source units.
            :type src: UnitsContainer
            :param dst: destination units.
            :type dst: UnitsContainer
    
            :return: converted value
            """
    
        if check_dimensionality:
    
            src_dim = self._get_dimensionality(src)
            dst_dim = self._get_dimensionality(dst)
    
            # If the source and destination dimensionality are different,
            # then the conversion cannot be performed.
            if src_dim != dst_dim:
>               raise DimensionalityError(src, dst, src_dim, dst_dim)
E               pint.errors.DimensionalityError: Cannot convert from 'meter ** 2 / second' ([length] ** 2 / [time]) to 'kilogram / meter / second' ([mass] / [length] / [time])

../../../../.local/lib/python3.6/site-packages/pint/registry.py:708: DimensionalityError
____ KValuesCalculationTest.test_k_value_super_thick_orifice_high_headloss _____

self = <tests.test_k_value_of_reductions_utility.KValuesCalculationTest testMethod=test_k_value_super_thick_orifice_high_headloss>

    def test_k_value_super_thick_orifice_high_headloss(self):
>       self.assertAlmostEqual(k.k_value_orifice(pipe.OD(6), pipe.OD(4), 60*u.inch, 1 * u.L / u.s), 1.8350488368427034)

tests/test_k_value_of_reductions_utility.py:42: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:172: in wrapper
    result = func(*new_values, **kw)
aide_design/shared/utility.py:136: in wrapper
    result = func(*args, **kwargs)
aide_design/shared/k_value_of_reductions_utility.py:188: in k_value_orifice
    k = k_value_reduction(id_pipe * u.m, id_orifice * u.m, flow * u.m**3/u.s) + k_value_expansion(id_orifice * u.m,
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:172: in wrapper
    result = func(*new_values, **kw)
aide_design/shared/utility.py:136: in wrapper
    result = func(*args, **kwargs)
aide_design/shared/k_value_of_reductions_utility.py:113: in k_value_reduction
    f = pc.fric(flow, id_entrance, NU, ROUGHNESS)
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:170: in wrapper
    new_values, values_by_name = converter(ureg, values, strict)
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:116: in _converter
    args_as_uc[ndx][0])
../../../../.local/lib/python3.6/site-packages/pint/registry.py:1214: in _convert
    return super(ContextRegistry, self)._convert(value, src, dst, inplace)
../../../../.local/lib/python3.6/site-packages/pint/registry.py:937: in _convert
    return super(NonMultiplicativeRegistry, self)._convert(value, src, dst, inplace)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pint.registry.UnitRegistry object at 0x10b3e65f8>, value = 1e-06
src = <UnitsContainer({'meter': 2.0, 'second': -1.0})>
dst = <UnitsContainer({'kilogram': 1.0, 'meter': -1.0, 'second': -1.0})>
inplace = False, check_dimensionality = True

    def _convert(self, value, src, dst, inplace=False, check_dimensionality=True):
        """Convert value from some source to destination units.
    
            :param value: value
            :param src: source units.
            :type src: UnitsContainer
            :param dst: destination units.
            :type dst: UnitsContainer
    
            :return: converted value
            """
    
        if check_dimensionality:
    
            src_dim = self._get_dimensionality(src)
            dst_dim = self._get_dimensionality(dst)
    
            # If the source and destination dimensionality are different,
            # then the conversion cannot be performed.
            if src_dim != dst_dim:
>               raise DimensionalityError(src, dst, src_dim, dst_dim)
E               pint.errors.DimensionalityError: Cannot convert from 'meter ** 2 / second' ([length] ** 2 / [time]) to 'kilogram / meter / second' ([mass] / [length] / [time])

../../../../.local/lib/python3.6/site-packages/pint/registry.py:708: DimensionalityError
_________________________ LfomTest.test_drillbit_area __________________________

self = <tests.test_lfom.LfomTest testMethod=test_drillbit_area>

    def test_drillbit_area(self):
        """drillbit_area should give known result with known input.
            Test cases were calculated using outputs from original Mathcad code and
            the nominal diameter function written in pipedatabase which has already
            been tested.
    
            """
        checks = ((60 * u.L/u.s, 20 * u.cm, mat.DIAM_DRILL_ENG, 0.0007669903939428206 * u.m**2),
                  (60 * u.L/u.s, 40 * u.cm, mat.DIAM_DRILL_ENG, 0.0007669903939428206 * u.m**2),
                  (20 * u.L/u.s, 20 * u.cm, mat.DIAM_DRILL_ENG, 3.141592653589793 * u.m**2),
                  (20 * u.L/u.s, 40 * u.cm, mat.DIAM_DRILL_ENG, 0.0007669903939428206 * u.m**2))
        for i in checks:
            with self.subTest(i=i):
>               self.assertAlmostEqual(lfom.drillbit_area(i[0], i[1], i[2]), i[3])
E               AssertionError: <Quantity(0.0015517916547360866, 'meter ** 2')> != <Quantity(0.0007669903939428206, 'meter ** 2')> within 7 places

tests/test_lfom.py:159: AssertionError
________________________ LfomTest.test_orifice_diameter ________________________

self = <tests.test_lfom.LfomTest testMethod=test_orifice_diameter>

    def test_orifice_diameter(self):
        FLOW = 31 * u.L / u.s
        HL_LFOM = 20 * u.cm
        drill_bits = np.arange(5, 25, 5) * u.mm
>       self.assertAlmostEqual(lfom.orifice_diameter(FLOW,HL_LFOM,drill_bits), 0.7874015748031495* u.m)
E       AssertionError: <Quantity(0.02, 'meter')> != <Quantity(0.7874015748031495, 'meter')> within 7 places

tests/test_lfom.py:168: AssertionError
______________________ FrictionFuncsTest.test_fric_units _______________________

self = <tests.test_physchem.FrictionFuncsTest testMethod=test_fric_units>

    def test_fric_units(self):
        """fric should handle units correctly."""
        base = pc.fric(100, 2, 0.001, 1)
        checks = ([100 * u.m**3/u.s, 2 * u.m, 0.001 * u.m**2/u.s, 1 * u.m],
                  [100000 * u.L/u.s, 2, 0.001, 1],
                  [100, 20 * u.dm, 0.001, 1],
                  [100, 2, 10 * u.cm**2/u.s, 1],
                  [100, 2, 0.001, 1000 * u.mm])
        for i in checks:
            with self.subTest(i=i):
>               self.assertAlmostEqual(pc.fric(*i), base)

tests/test_physchem.py:283: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:170: in wrapper
    new_values, values_by_name = converter(ureg, values, strict)
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:116: in _converter
    args_as_uc[ndx][0])
../../../../.local/lib/python3.6/site-packages/pint/registry.py:1214: in _convert
    return super(ContextRegistry, self)._convert(value, src, dst, inplace)
../../../../.local/lib/python3.6/site-packages/pint/registry.py:937: in _convert
    return super(NonMultiplicativeRegistry, self)._convert(value, src, dst, inplace)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pint.registry.UnitRegistry object at 0x10b3e65f8>, value = 0.001
src = <UnitsContainer({'meter': 2.0, 'second': -1.0})>
dst = <UnitsContainer({'kilogram': 1.0, 'meter': -1.0, 'second': -1.0})>
inplace = False, check_dimensionality = True

    def _convert(self, value, src, dst, inplace=False, check_dimensionality=True):
        """Convert value from some source to destination units.
    
            :param value: value
            :param src: source units.
            :type src: UnitsContainer
            :param dst: destination units.
            :type dst: UnitsContainer
    
            :return: converted value
            """
    
        if check_dimensionality:
    
            src_dim = self._get_dimensionality(src)
            dst_dim = self._get_dimensionality(dst)
    
            # If the source and destination dimensionality are different,
            # then the conversion cannot be performed.
            if src_dim != dst_dim:
>               raise DimensionalityError(src, dst, src_dim, dst_dim)
E               pint.errors.DimensionalityError: Cannot convert from 'meter ** 2 / second' ([length] ** 2 / [time]) to 'kilogram / meter / second' ([mass] / [length] / [time])

../../../../.local/lib/python3.6/site-packages/pint/registry.py:708: DimensionalityError
___________________ DiamFuncsTest.test_diam_pipemajor_units ____________________

self = <tests.test_physchem.DiamFuncsTest testMethod=test_diam_pipemajor_units>

    def test_diam_pipemajor_units(self):
        """diam_pipemajor should handle units correctly."""
        base = pc.diam_pipemajor(1, 2, 0.03, 0.004, 0.005).magnitude
        checks = ([1 * u.m**3/u.s, 2 * u.m, 0.03 * u.m,
                   0.004 * u.m**2/u.s, 0.005 * u.m],
                  [1000 * u.L/u.s, 2, 0.03, 0.004, 0.005],
                  [1, 0.002 * u.km, 0.03, 0.004, 0.005],
                  [1, 2, 3 * u.cm, 0.004, 0.005],
                  [1, 2, 0.03, 40 * u.cm**2/u.s, 0.005],
                  [1, 2, 0.03, 0.004, 5 * u.mm])
        for i in checks:
            with self.subTest(i=i):
>               self.assertAlmostEqual(pc.diam_pipemajor(*i).magnitude, base)

tests/test_physchem.py:993: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:170: in wrapper
    new_values, values_by_name = converter(ureg, values, strict)
../../../../.local/lib/python3.6/site-packages/pint/registry_helpers.py:116: in _converter
    args_as_uc[ndx][0])
../../../../.local/lib/python3.6/site-packages/pint/registry.py:1214: in _convert
    return super(ContextRegistry, self)._convert(value, src, dst, inplace)
../../../../.local/lib/python3.6/site-packages/pint/registry.py:937: in _convert
    return super(NonMultiplicativeRegistry, self)._convert(value, src, dst, inplace)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pint.registry.UnitRegistry object at 0x10b3e65f8>, value = 0.004
src = <UnitsContainer({'meter': 2.0, 'second': -1.0})>
dst = <UnitsContainer({'kilogram': 1.0, 'meter': -1.0, 'second': -1.0})>
inplace = False, check_dimensionality = True

    def _convert(self, value, src, dst, inplace=False, check_dimensionality=True):
        """Convert value from some source to destination units.
    
            :param value: value
            :param src: source units.
            :type src: UnitsContainer
            :param dst: destination units.
            :type dst: UnitsContainer
    
            :return: converted value
            """
    
        if check_dimensionality:
    
            src_dim = self._get_dimensionality(src)
            dst_dim = self._get_dimensionality(dst)
    
            # If the source and destination dimensionality are different,
            # then the conversion cannot be performed.
            if src_dim != dst_dim:
>               raise DimensionalityError(src, dst, src_dim, dst_dim)
E               pint.errors.DimensionalityError: Cannot convert from 'meter ** 2 / second' ([length] ** 2 / [time]) to 'kilogram / meter / second' ([mass] / [length] / [time])

../../../../.local/lib/python3.6/site-packages/pint/registry.py:708: DimensionalityError
==================== 10 failed, 137 passed in 2.01 seconds =====================

ethan92429 commented on aide_design/designs/civil_work/ent_tank.py:
```diff
@@ -0,0 +1,40 @@
+        L_plate
+    )
+
+    drain_OD = staticmethod(drain_OD)
```
Any opinions about this? What behavior would you like to see?

oliver-leung commented:
Closing as we are moving over to a class-based framework for our design code.
